src/transformers/models/llava/processing_llava.py

```diff
@@ -0,0 +1,220 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processor class for LlavaModel
+"""
+
+import os
+from typing import List, Optional, Union
+
+from ...image_processing_utils import BatchFeature
+from ...image_utils import ImageInput
+from ...processing_utils import ProcessorMixin
+from ...tokenization_utils_base import PaddingStrategy, TextInput, TruncationStrategy
+from ...utils import (
+    TensorType,
+    is_torch_available,
+    is_vision_available,
+)
+
+
+if is_vision_available():
+    from PIL import Image
+
+if is_torch_available():
+    import torch
+
+    from ..clip import CLIPVisionModel
+
+
+class LlavaProcessor(ProcessorMixin):
+    r"""
+    Constructs an LLava processor which wraps a CLIP image processor, CLIP vision model and a LLaMa/T5 tokenizer into a
+    single processor.
+
+    [`LlavaProcessor`] offers all the functionalities of [`CLIPImageProcessor`] and [`AutoTokenizer`]. See the
+    docstring of [`~BlipProcessor.__call__`] and [`~BlipProcessor.decode`] for more information.
+
+    Args:
+        image_processor (`CLIPProcessor`):
+            An instance of [`CLIPProcessor`]. The image processor is a required input.
+        tokenizer (`AutoTokenizer`):
+            An instance of ['PreTrainedTokenizer`]. The tokenizer is a required input.
+        vision_model (`CLIPVisionModel`):
+            An instance of ['CLIPVisionModel`]. The Vision Model is a required input.
+    """
+    attributes = ["image_processor", "tokenizer"]
+    tokenizer_class = "AutoTokenizer"
+    image_processor_class = "CLIPImageProcessor"
+
+    def __init__(self, image_processor, tokenizer, vision_model):
+        super().__init__(image_processor, tokenizer)
+
+        self.vision_model = vision_model
+        self.DEFAULT_IMAGE_TOKEN = "<image>"
+        self.IMAGE_TOKEN_INDEX = -200
```

Comment:
from PEP: "Constants are usually defined on a module level and written in all capital letters with underscores separating words."
So, it would be better to have it as a module constant as done in IDEFICS [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/idefics/processing_idefics.py#L32)